pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -515,7 +535,7 @@ func (c *Cluster) startMon(m *monConfig, hostname string) error {
 			return fmt.Errorf("failed to create mon deployment %s. %+v", m.ResourceName, err)
 		}
 		logger.Infof("deployment for mon %s already exists. updating if needed", m.ResourceName)
-		if _, err := updateDeploymentAndWait(c.context, d, c.Namespace); err != nil {
+		if err := updateDeploymentAndWait(c.context, d, c.Namespace, c.clusterInfo.Name, c.clusterInfo.CephVersion); err != nil {
```

Comment:
The `CephVersion` provided here needs to be the current running version of Ceph for when it is later passed to `OkToStop` and checked with `IsAtLeast`.  If you are upgrading from a pre 14.2.1 version to a version 14.2.1 or newer, `IsAtLeast` will incorrectly return `true` and then fail trying to run `mon ok-to-stop` on a mon pod running a pre 14.2.1 version.  There are several more calls to this function using `c.clusterInfo.CephVersion` for other services that likely need updated as well.

Reply:
You're right, it might get tricky though if it's a second orchestration upgrade and we have daemons with a different version...